Replace clear() method for backward compatibility.

n0trax opened the pull request:
##### SUMMARY
The clear() method is only supported from python >= 3.3, but acme_certificate has python >=2.6 as requirement. Therefore we should replace the clear() method.

Fixes #64501 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
acme_certificate

n0trax committed:
Replace clear() method for backward compatibility.

felixfontein commented:
Thanks for spotting this! Unfortunately the integration tests didn't catch this since they don't use ACME v1... BTW, out of curiosity: why are you still using ACME v1? Is it because you didn't upgrade yet, or because the ACME server you're using doesn't support ACME v2?

In any case, you need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to):
```
bugfixes:
- "acme_certificate - fix crash when module is used with Python 2.x."
```

felixfontein commented:
(In case you're interested, [Let's Encrypt's ACME v1 deprecation](https://community.letsencrypt.org/t/end-of-life-plan-for-acmev1/88430).)

MarkusTeufelberger commented:
lgtm

n0trax commented:
@felixfontein the task uses the ACMEv1 because it is an older playbook and the default acme_version of the acme_certificate module is still ACMEv1.
But thanks for the hint :-) I updated my letsEncrypt playbook to use ACMEv2.

I also added the changelog fragment.

felixfontein reviewed (approved):
shipit

felixfontein commented:
@n0trax thanks for reporting and fixing this! I'll create backport PRs so this will also get fixed in 2.8.x and 2.9.x eventually.
@MarkusTeufelberger thanks for reviewing!

@n0trax actually the default for `acme_version` will change eventually; from Ansible 2.10 on you'll get a warning if you don't specify it, with a note that there will no longer be a default for Ansible 2.14. From then on, you'll have to specify the version explicitly.

felixfontein commented:
(Just noticed, no backport necessary to stable-2.8 since that doesn't happen in Ansible 2.8.x. The broken code was new for 2.9.0.)
